exonum-light-client/src/main/java/com/exonum/client/ExonumHttpClient.java

```diff
@@ -166,7 +167,10 @@ public BlockResponse getBlockByHeight(long height) {
 
       blocks.addAll(blocksResponse.getBlocks());
 
-      rangeLast = blocksResponse.getBlocksRangeStart() - 1;
+      long blocksRangeStart = blocksResponse.getBlocksRangeStart();
+      checkState(blocksRangeStart <= rangeLast,
```

Comment:
This code distrusts the Exonum API, doesn't it?

Reply:
Why does it distract? It checks theoretically impossible case to prevent forever loop.

Reply:
If API is hacked this library might have even bigger problems, isn't it?

Reply:
definitely yes. Do you think this check is redundant?

Reply:
If we assume we can rely on Exonum returning proper responses, then, probably, yes.